File: tests/typesafety/typescript/typed_arguments_misuse.ts
// expect: fails

function orderLine(name: string, unitPrice: number, quantity: number, gift: boolean): string {
  const note = gift ? " (gift)" : "";
  return `${name} x${quantity} at ${unitPrice.toFixed(2)}${note}`;
}

export const line = orderLine(3, "tea", true, 1.95); // rejected by the checker

File: tests/typesafety/python/typed_arguments_misuse.py
# expect: fails
# title: The mixed-up call, rejected by the checker
# misuse-of: typed_arguments
def invoice_line(description: str, price_pence: int, quantity: int, taxed: bool) -> str:
    note = " +tax" if taxed else ""
    return f"{description} x{quantity} at {price_pence}d{note}"


line = invoice_line(80, "handlebar grip", True, 2)
